core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -482,8 +482,19 @@ private boolean processSyncPath(AlluxioURI path) {
     if (path == null) {
       return false;
     }
+
+    boolean hasCache;
+    try {
+      hasCache = (mStatusCache.getStatus(path) != null);
+    } catch (FileNotFoundException e) {
+      LogUtils.warnWithException(LOG, "Failed to process sync path: {}", path, e);
+      return false;
+    }
+
     LockingScheme scheme;
-    if (mForceSync) {
+    // hasCache is true means listStatus already prefetched metadata of children,
+    // update metadata for such cases
+    if (mForceSync || hasCache) {
       scheme = new LockingScheme(path, LockPattern.READ, true);
```

Comment:
Because listStatus already prefetched the metadata of children, we can update children's metadata without accessing ufs once again for each child. If `sync=true`, alluxio
1. Update the metadata directly if it is already in cache
2. Access ufs to sync get the latest metadata if it is not in cache